app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,13 @@
+class RecipesController < ApplicationController
+  def show
+    @recipe = Recipe.find(params[:id])
+    @total_cost = @recipe.total_cost
+  end
+
+  def update
+    ingredient = Ingredient.find(params[:ingredient_id])
+    Recipe.first.ingredients.create!(name: ingredient.name, cost: ingredient.cost)
```

Comment:
Instead of creating the joins record like this, try:
`recipe = Recipe.find(params[:recipe_id]`
`ingredient = Ingredient.find(params[:ingredient_id])`
`RecipeIngredient.create!(recipe_id: recipe.id, ingredient_id: ingredient.id)`

Also, it generally more RESTful to create a joins table association in a `create` action, and you could direct it to a different controller called `recipe_ingredients_controller.rb`. That indicates to other developers that you're using that create action to create a relationship between those two resources: ingredients and recipe. Using an `update` like you did her is slightly less restful because that `update` is generally thought  to be updating  just the `recipe` object's attributes rather than creating a joins record. This is just a slight infraction though